### PR TITLE
fix(php): add function return type

### PR DESCRIPTION
**Pull request by vargenau**
## Description

Functions have no return type.

### Changes

Add function return type.

### CODE REVIEW

File: src/lib/php/common-menu.php
Comment (GMishx) [inline src/lib/php/common-menu.php:1]: Can add return type `:string` to `MenuPage()`.

Comment (GMishx) [reply src/lib/php/common-menu.php:1]: Type `:string` to `MenuEndlessPage()`.

File: src/lib/php/common-parm.php
Comment (GMishx) [inline src/lib/php/common-parm.php:1]: Can add return type `:string` to `Traceback()`.

Comment (GMishx) [reply src/lib/php/common-parm.php:1]: Type `:string` to `Traceback_uri()`.

Comment (GMishx) [reply src/lib/php/common-parm.php:1]: Type `:string` to `Traceback_parm()`.

Comment (GMishx) [reply src/lib/php/common-parm.php:1]: Type `:string` to `Traceback_dir()`.

File: src/lib/php/common-perms.php [hunk 1]
```diff
@@ -147,7 +147,7 @@ function GetGroupUsers($user_pk, $group_pk, &$GroupArray)
    *  -  [group_name]
    *  -  [group_perm]
    **/
-function GetUsersGroups($user_pk='')
+function GetUsersGroups($user_pk=''): array
```
Comment (GMishx) [inline src/lib/php/common-perms.php:150]: The function does not always return. So, the value can also be null.
```suggestion
function GetUsersGroups($user_pk=''): ?array
```

File: src/lib/php/common-perms.php
Comment (GMishx) [inline src/lib/php/common-perms.php:1]: Can add return type `: ?string` to `DeleteGroup()`.

File: src/lib/php/common-plugin.php
Comment (GMishx) [inline src/lib/php/common-plugin.php:1]: Can set the return type to `:string|int` for `plugin_find_id()`.

Comment (GMishx) [reply src/lib/php/common-plugin.php:1]: Type `:?Plugin` to `plugin_find()`.

File: src/lib/php/common-scheduler.php
Comment (GMishx) [inline src/lib/php/common-scheduler.php:1]: Can add type `: false|string` to `fo_scheduler_read()`.

Comment (GMishx) [reply src/lib/php/common-scheduler.php:1]: Type `: false|int` to `fo_scheduler_write()`.

Comment (GMishx) [reply src/lib/php/common-scheduler.php:1]: Type `: ?array` to `GetRunnableJobList()`.

File: src/lib/php/common-string.php
Comment (GMishx) [inline src/lib/php/common-string.php:1]: Can add type `: string` to `convertToUTF8()`.

Comment (GMishx) [reply src/lib/php/common-string.php:1]: Type `: false|string` to `tryConvertToUTF8()`.

File: src/lib/php/common-sysconfig.php
Comment (GMishx) [inline src/lib/php/common-sysconfig.php:1]: Can add type `: void` to `Populate_sysconfig()`.

Comment (GMishx) [reply src/lib/php/common-sysconfig.php:1]: Type `: int` to `check_IP()`.

Comment (GMishx) [reply src/lib/php/common-sysconfig.php:1]: Type `: void` to `set_python_path()`.

File: src/lib/php/common-tags.php
Comment (GMishx) [inline src/lib/php/common-tags.php:1]: Can add return type `: array` to `GetAllTags()`.

File: src/lib/php/common-ui.php
Comment (GMishx) [inline src/lib/php/common-ui.php:1]: Can add type `: true|string` to `DownloadString2File()`.

Comment (GMishx) [reply src/lib/php/common-ui.php:1]: Type `: string` to `GetUploadtreeTableName()`.

Comment (GMishx) [reply src/lib/php/common-ui.php:1]: Type `: string` to `GetUploadName()`.

File: src/lib/php/fossdash-config.php
Comment (GMishx) [inline src/lib/php/fossdash-config.php:1]: Can add type `: int` to `check_cron_job_inteval()`.

Comment (GMishx) [reply src/lib/php/fossdash-config.php:1]: Type `: int` to `check_fossology_instance_name()`.

Comment (GMishx) [reply src/lib/php/fossdash-config.php:1]: Type `: int` to `check_fossdash_cleaning()`.

Comment (GMishx) [reply src/lib/php/fossdash-config.php:1]: Type `: int` to `check_username()`.

Comment (GMishx) [reply src/lib/php/fossdash-config.php:1]: Type `: int` to `check_password()`.

**Review by GMishx — CHANGES_REQUESTED** [review]

**Comment by vargenau** [comment]
@GMishx 
Hi,
I have implemented the changes you requested, but now I have some tests failing, and I do not understand what the errors are.

**Pull request closed without merging**